pkg/virtual/syncer/builder/forwarding.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	"context"
+
+	"github.com/kcp-dev/logicalcluster"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apiextensions-apiserver/pkg/registry/customresource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kube-openapi/pkg/validation/validate"
+
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/apiserver"
+	registry "github.com/kcp-dev/kcp/pkg/virtual/framework/forwardingregistry"
+)
+
+var _ registry.ClientGetter = (*clusterAwareClientGetter)(nil)
+
+type clusterAwareClientGetter struct {
```

Comment:
that's a strange construct. We have the dynamic cluster client that is basically the same. Following "compose as far up as possible and avoid intermediate constructs", this should be used and composed with `genericapirequest.ValidClusterFrom` where it is used.

Reply:
Removed and replaced with `dynamic.ClusterInterface`.